src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -346,14 +343,7 @@ private HttpClient getHttpClient(PrintStream logger, AbstractBuild<?, ?> build)
         String stashServer = stashServerBaseUrl;
         DescriptorImpl descriptor = getDescriptor();
 
-		// Determine if we are using the local or global settings
-		String credentialsId = getCredentialsId();
-		if (StringUtils.isBlank(credentialsId)) {
-			credentialsId = descriptor.getCredentialsId();
-		}
-
-		Credentials credentials = CredentialsMatchers.firstOrNull(CredentialsProvider.lookupCredentials(CertificateCredentials.class,
-				Jenkins.getInstance(), ACL.SYSTEM), CredentialsMatchers.withId(credentialsId));
+        CertificateCredentials certificateCredentials = getCredentials(CertificateCredentials.class, build.getProject());
```

Comment:
I'll have another look at it after work this evening.

But I don't really see how the credentialId's were different before my refactoring? 
Maybe the difference is to be found in the "strange" assignment of the credentialsId member variable in the username/password credentials lookup block, which I didn't fully understand?

But maybe it was just the late hour..